### ISSUE TITLE
Patch to handle shards greater than 100

### ISSUE THREAD

**Issue by sjatlyft** [qubole/kinesis-sql#29]
https://github.com/qubole/kinesis-sql/blob/caa88fb7d8b5a0f3aab2472a8c554d2720401b16/src/main/scala/org/apache/spark/sql/kinesis/KinesisReader.scala#L173

[patchToHandleMoreShards.patch.gz](https://github.com/qubole/kinesis-sql/files/2507306/patchToHandleMoreShards.patch.gz)

**Comment by itsvikramagr** [comment]
@sjatlyft - Thanks for the patch. Do you want to start a PR for this?

**Comment by itsvikramagr** [comment]
@Vinaypandit - How often do you think you are going to do re-sharding? DescribeStream API call is used to figure out merged and split shards. if re-sharding is not so often, you would want to use the higher value of `describeShardInterval` (say 1 hr)

If your re-sharding is often, to prevent the API rate limiting, you would like to increase the higher value of `kinesis.client.numRetries` & `kinesis.client.retryIntervalMs` 

In short, the streaming code has already taken care of the AWS rate limit to a great extent.

**Comment by Vinaypandit** [comment]
@itsvikramagr - We will not be resharding  that often so I will increase that value and give it a shot. Thanks for your help.
